src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+  const handleOnKeyDown = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      const [minVal, maxVal] = getAllowableThumbRange(
+        thumbMode,
+        values,
+        index,
+        railDomain,
+        step,
+      );
+
+      if (event.key === 'ArrowLeft') {
+        const newValue = Math.max(values[index] - step, minVal);
+        setThumbValue(index, newValue);
+      } else if (event.key === 'ArrowRight') {
+        const newValue = Math.min(values[index] + step, maxVal);
+        setThumbValue(index, newValue);
+      } else if (event.key === 'Home') {
+        setThumbValue(index, minVal);
+      } else if (event.key === 'End') {
+        setThumbValue(index, maxVal);
+      }
+    },
+    [thumbMode, values, index, railDomain, step, setThumbValue],
+  );
+
+  const handlePointerDownOnThumb = React.useCallback(
+    (event: React.PointerEvent) => {
+      thumbRef.current?.focus();
+      setThumbMoveIndex(index);
+      event.preventDefault();
+      event.stopPropagation();
+    },
+    [index, setThumbMoveIndex],
+  );
+
+  const [hasFocus, setHasFocus] = React.useState(false);
+  const [isHovered, setIsHovered] = React.useState(false);
+
+  const generateTooltip = React.useCallback(
+    (val: number): string => {
+      return toolTipFunction
+        ? toolTipFunction(val, step)
+        : formatNumberValue(val, step);
+    },
+    [toolTipFunction, step],
+  );
+
+  const currentValue = values[index];
+  const tooltipContent = generateTooltip(currentValue);
+  const leftPercent =
+    (100.0 * (currentValue - railDomain.min)) /
+    (railDomain.max - railDomain.min);
+
+  const thumbClassNames = React.useMemo(
+    () => cx('iui-slider-thumb', isActive && 'iui-active'),
+    [isActive],
+  );
+  const showTooltip = !hideTooltip && (isActive || hasFocus || isHovered);
+  return (
+    <Tooltip
+      visible={showTooltip}
+      content={tooltipContent}
+      placement={toolTipPosition}
+    >
+      <div
+        ref={thumbRef}
+        style={{ left: `${leftPercent}%` }}
+        className={thumbClassNames}
+        role='slider'
+        tabIndex={0}
+        aria-valuemin={railDomain.min}
+        aria-valuenow={railDomain.max}
+        aria-valuemax={currentValue}
+        onPointerDown={handlePointerDownOnThumb}
+        onKeyDown={handleOnKeyDown}
+        onFocus={() => setHasFocus(true)}
+        onBlur={() => setHasFocus(false)}
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+      />
+    </Tooltip>
+  );
+};
+
+/** Properties for [[Slider]] component */
+export type SliderProps = {
+  /**
+   * Set focus on first thumb in slider element.
+   * @default false
+   */
+  setFocus?: boolean;
+  /**
+   * Allowable range of values.
+   */
+  railDomain: ValueDomain;
+  /**
+   * Array of one or more values to show.
+   */
+  values: number[];
+  /**
+   *  Determines which segments are shown with color.
```

Comment:
I think "segments" is not very clear (and neither is track). Could clarify what part this prop actually changes

Reply:
I am still using "segments" but I am open to another name if you have a more descriptive one.